proof_test.go

```diff
@@ -0,0 +1,120 @@
+package merkletree
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+)
+
+func computeLeafHash(ap *AuthenticationPath) (leafHash []byte) {
+	leaf := ap.Leaf()
+	if leaf.Commitment() != nil {
+		// user leaf node
+		leafHash = crypto.Digest(
+			[]byte{LeafIdentifier},              // K_leaf
+			[]byte(ap.TreeNonce()),              // K_n
+			[]byte(ap.Index()),                  // i
+			[]byte(util.IntToBytes(ap.Level())), // l
+			[]byte(leaf.Commitment()),           // commit(key|| value)
+		)
+	} else {
+		// empty leaf node
+		leafHash = crypto.Digest(
+			[]byte{EmptyBranchIdentifier},       // K_empty
+			[]byte(ap.TreeNonce()),              // K_n
+			[]byte(ap.Index()),                  // i
+			[]byte(util.IntToBytes(ap.Level())), // l
+		)
+	}
+	return
+}
+
+func computeProofHash(ap *AuthenticationPath, depth int) (leftHash, rightHash []byte) {
```

Comment:
This seems to be recomputing the proof hash from the root to the leaf (as depth is increasing in the recursion), but it should be from the leaf up to the root. This way, the client can compare the recomputed root from the auth path with the root in the STR for the same epoch.